src/object/srv_ec_aggregate.c

```diff
@@ -2400,6 +2400,13 @@ agg_filter(daos_handle_t ih, vos_iter_desc_t *desc, void *cb_arg, unsigned int *
 		*acts = VOS_ITER_CB_SKIP;
 		goto done;
 	}
+
+	/* This MUST be the last check */
+	if (desc->id_type == VOS_ITER_OBJ && vos_bkt_iter_skip(ih, desc)) {
+		agg_param->ap_credits++;
+		*acts |= VOS_ITER_CB_SKIP;
+		goto done;
```

Comment:
If it must be the last check, then do not need "goto done".

Reply:
Well, I was just thinking that some other operations (not related to skip check) could be added in the future. I can remove it if the patch needs be refreshed.